c7n/resources/ssm.py

```diff
@@ -597,3 +598,68 @@ def register_resource(cls, registry, resource_class):
 
 
 resources.subscribe(PostItem.register_resource)
+
+
+@resources.register('ssm-data-sync')
+class SSMDataSync(QueryResourceManager):
+    """Resource for AWS DataSync
+    https://docs.aws.amazon.com/systems-manager/latest/userguide/sysman-inventory-datasync.html
+    """
+    class resource_type(TypeInfo):
+
+        enum_spec = ('list_resource_data_sync', 'ResourceDataSyncItems', None)
+        service = 'ssm'
+        arn_type = 'datasync'
+        id = 'DataSync'
+        name = 'Title'
+
+    permissions = ('ssm:ListResourceDataSync',)
+
+
+@SSMDataSync.filter_registry.register('kms-key')
+class KmsFilter(KmsRelatedFilter):
+    """
+    Filter a resource by its associcated kms key and optionally the aliasname
```

Comment:
can we skip this docstring, we have another pr which is nuking the extant so the baseclass docs can be utilized/updated in a single place.

Reply:
Got it.